### PR TITLE
Update to Xcode 15.1

### PR DESCRIPTION
**Pull request by jaceklyp**
Task/Issue URL: https://app.asana.com/0/414709148257752/1205555242718448/f

Description:
Update Xcode to version 15.0.0

Steps to test this PR:

Make sure all tests pass

### CODE REVIEW

File: DuckDuckGoTests/NetworkProtectionStatusViewModelTests.swift [hunk 1]
```diff
@@ -204,6 +204,6 @@ final class NetworkProtectionStatusViewModelTests: XCTestCase {
             condition()
         }
         let expectation = XCTNSPredicateExpectation(predicate: predicate, object: nil)
-        wait(for: [expectation], timeout: 5)
+        wait(for: [expectation], timeout: 20)
```
Comment (samsymons) [inline DuckDuckGoTests/NetworkProtectionStatusViewModelTests.swift:207]: Now that the simulator issue has been fixed, we can consider reverting these timeout changes. Hopefully they'll no longer be necessary.

**Review by samsymons — APPROVED** [review]
This looks good to me, thanks!

@jaceklyp I left one comment about reverting the timeout changes that were added back when the simulator performance was low.

Also if you could give the branch another try locally to make sure it looks good to you after all the main branch changes that have been merged, that would be great.

File: .github/workflows/alpha.yml [hunk 2]
```diff
@@ -76,7 +76,7 @@ jobs:
 
       # Using Xcode 15 as the alpha build uses iOS 17 APIs
     - name: Select Xcode
-      run: sudo xcode-select -s /Applications/Xcode_15.0.1.app/Contents/Developer
+      run: sudo xcode-select -s /Applications/Xcode_$(<.xcode-version)/Contents/Developer
```
Comment (ayoy) [inline .github/workflows/alpha.yml]: @jaceklyp you've missed the .app extension here:
```suggestion
      run: sudo xcode-select -s /Applications/Xcode_$(<.xcode-version).app/Contents/Developer
```

Comment (samsymons) [reply .github/workflows/alpha.yml]: I believe that was my mistake, @jaceklyp is innocent 😇 Fixed, thanks!

**Review by ayoy — APPROVED** [review]

**Pull request merged**